src/inoreader_api.cpp

```diff
@@ -0,0 +1,300 @@
+
+	utils::set_common_curl_options(handle, cfg);
+	curl_easy_setopt(handle, CURLOPT_WRITEFUNCTION, my_write_data);
+	curl_easy_setopt(handle, CURLOPT_WRITEDATA, &result);
+	curl_easy_setopt(handle, CURLOPT_URL, INOREADER_SUBSCRIPTION_LIST);
+	curl_easy_perform(handle);
+	curl_easy_cleanup(handle);
+	curl_slist_free_all(custom_headers);
+
+	LOG(level::DEBUG, "inoreader_api::get_subscribed_urls: document = %s", result);
+
+	json_object* reply = json_tokener_parse(result.c_str());
+	if (reply == nullptr) {
+		LOG(level::ERROR, "inoreader_api::get_subscribed_urls: failed to parse response as JSON.");
+		return urls;
+	}
+
+
+	json_object* subscription_obj {};
+	json_object_object_get_ex(reply, "subscriptions", &subscription_obj);
+	struct array_list * subscriptions = json_object_get_array(subscription_obj);
+
+	int len = array_list_length(subscriptions);
+
+	for (int i=0; i<len; i++) {
+		std::vector<std::string> tags;
+		json_object* sub = json_object_array_get_idx(subscription_obj, i);
+
+		json_object* node {};
+
+		json_object_object_get_ex(sub, "id", &node);
+		const char * id = json_object_get_string(node);
+
+		json_object_object_get_ex(sub, "title", &node);
+		const char * title = json_object_get_string(node);
+
+		// Ignore URLs where ID start with given prefix - those never load,
+		// always returning 404 and annoying people
+		const char* prefix = "tor/sponsored/";
+		if (strncmp(id, prefix, strlen(prefix)) != 0) {
+			tags.push_back(std::string("~") + title);
+
+			json_object_object_get_ex(sub, "categories", &node);
+			struct array_list * categories = json_object_get_array(node);
+			for (int i = 0; i < array_list_length(categories); i++) {
+				json_object* cat = json_object_array_get_idx(node, i);
+				json_object* label_node {};
+				json_object_object_get_ex(cat, "label", &label_node);
+				const char * label = json_object_get_string(label_node);
+				tags.push_back(std::string(label));
+			}
+
+			auto url = strprintf::fmt(
+			               "%s%s?n=%u",
+			               INOREADER_FEED_PREFIX,
+			               id,
+			               cfg->get_configvalue_as_int("inoreader-min-items"));
+			urls.push_back(tagged_feedurl(url, tags));
+		}
+	}
+
+	json_object_put(reply);
+
+	return urls;
+}
+
+void inoreader_api::add_custom_headers(curl_slist** custom_headers) {
+	if (auth_header.empty()) {
+		auth_header = strprintf::fmt("Authorization: GoogleLogin auth=%s", auth);
+	}
+	LOG(level::DEBUG, "inoreader_api::add_custom_headers header = %s", auth_header);
+	*custom_headers = curl_slist_append(*custom_headers, auth_header.c_str());
+	*custom_headers = curl_slist_append(*custom_headers, "AppId: 1000000588");
+	*custom_headers = curl_slist_append(*custom_headers, "AppKey: qRf9ZewWGwe630w61borYEtraeZMsMyS");
+}
+
+bool inoreader_api::mark_all_read(const std::string& feedurl) {
+	std::string real_feedurl = feedurl.substr(strlen(INOREADER_FEED_PREFIX), feedurl.length() - strlen(INOREADER_FEED_PREFIX));
```

Comment:
Same as earlier comment: no need for second argument here, `substr` got a useful default.